modules/api/test/LorisApiInstrumentsTest.php

```diff
@@ -0,0 +1,933 @@
+        );
+        $this->assertArrayHasKey(
+            'Validity',
+            $InstrArray['Flags']
+        );
+    }
+
+    /**
+     * Tests the HTTP PATCH request for the
+     * endpoint /projects/{project}/instruments{instrument}/flag
+     *
+     * @return void
+     */
+    public function testPatchCandidatesCandidVisitInstrumentsInstrumentFlags(): void
+    {
+        $candid     = '300003';
+        $visit      = 'V1';
+        $instrument = 'aosi';
+        $json       = [
+            'Meta'  => [
+                'Candidate'  => $candid,
+                'Visit'      => $visit,
+                'DDE'        => false,
+                'Instrument' => $instrument
+            ],
+            'Flags' => [
+                'Data_entry' => "2"
+            ]
+        ];
+        $response   = $this->client->request(
+            'PATCH',
+            "candidates/$candid/$visit/instruments/$instrument/flags",
+            [
+                'http_errors' => false,
+                'headers'     => $this->headers,
+                'json'        => $json
+            ]
+        );
+        if ($response->getStatusCode() === 404) {
+            $this->markTestIncomplete(
+                "Endpoint not found: PATCH" .
+                "candidates/$candid/$visit/instruments/$instrument/flags"
+            );
+        }
+        if ($response->getStatusCode() === 405) {
+            $this->markTestSkipped(
+                "Error 405"
+            );
+        }
+        $this->assertEquals(204, $response->getStatusCode());
+        // Verify the endpoint has a body
+        $body = $response->getBody();
+        $this->assertNotEmpty($body);
+    }
+
+    /**
+     * Tests the HTTP PUT request for the
+     * endpoint /projects/{project}/instruments{instrument}
+     *
+     * @return void
+     */
+    public function testPutCandidatesCandidVisitInstrumentsInstrumentFlags(): void
+    {
+        $candid     = '300004';
+        $visit      = 'V1';
+        $instrument = 'aosi';
+        $json       = [
+            'Meta'  => [
+                'Candidate'  => $candid,
+                'Visit'      => $visit,
+                'DDE'        => false,
+                'Instrument' => $instrument
+            ],
+            'Flags' => [
+                'Validity' => "2"
+            ]
+        ];
+        $response   = $this->client->request(
+            'PUT',
+            "candidates/$candid/$visit/instruments/$instrument/flags",
+            [
+                'http_errors' => false,
+                'headers'     => $this->headers,
+                'json'        => $json
+            ]
+        );
+        if ($response->getStatusCode() === 404) {
+            $this->markTestIncomplete(
+                "Endpoint not found: PUT" .
+                "candidates/$candid/$visit/instruments/$instrument/flags"
+            );
+        }
+        if ($response->getStatusCode() === 405) {
+            $this->markTestSkipped(
```

Comment:
The test should fail in this case.
It will be handled by the assettEquals(204 ...

Reply:
again it was not necessary, removing